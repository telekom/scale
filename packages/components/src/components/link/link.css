/**
* @prop --link-display: Display of the link
* @prop --link-flex-direction: Flex direction of the link
* @prop --link-align-items: Align items of the link
* @prop --link-justify-content: Justify content of the link
* @prop --link-position: Position of the link
* @prop --link-text-decoration: Text decoration of the link
* @prop --link-outline: Outline of the link
* @prop --link-cursor: Cursor of the link
* @prop --link-padding: Padding of the link
* @prop --link-font-size: Font size of the link
* @prop --link-font-weight: Font weight of the link
* @prop --link-color: Color of the link
* @prop --link-variant-primary-color: Color of the link variant primary
* @prop --link-variant-success-color: Color of the link variant success
* @prop --link-variant-warning-color: Color of the link variant warning
* @prop --link-variant-danger-color: Color of the link variant danger
* @prop --link-variant-info-color: Color of the link variant info
* @prop --link-hover-with-underline: Color of the link variant primary
*
*/
.link {
  display: var(--link-display, inline-flex);
  flex-direction: var(--link-flex-direction, row);
  align-items: var(--link-align-items, center);
  justify-content: var(--link-justify-content, center);
  position: var(--link-position, relative);
  text-decoration: var(--link-text-decoration, none);
  outline: var(--link-outline, none);
  cursor: var(--link-cursor, pointer);
  padding: var(--link-padding, 0);
  font-size: var(--link-font-size, 14px);
  font-weight: var(--link-font-weight, 500);
  color: var(--link-color, #000);
}

.link--disabled {
  cursor: var(--link-disabled-cursor, not-allowed);
  opacity: var(--link-disabled-opacity, 0.5);
}

.link--variant-primary {
  color: var(--link-variant-primary-color, #409eff);
}

.link--variant-success {
  color: var(--link-variant-success-color, #67c23a);
}

.link--variant-warning {
  color: var(--link-variant-warning-color, #e6a23c);
}

.link--variant-danger {
  color: var(--link-variant-danger-color, #f56c6c);
}

.link--variant-info {
  color: var(--link-variant-info-color, #909399);
}

.link--underline:hover {
  text-decoration: var(--link-hover-with-underline, underline);
}
