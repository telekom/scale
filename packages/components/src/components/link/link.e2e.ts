import { newE2EPage } from '@stencil/core/testing';

describe('t-link', () => {
  it('renders', async () => {
    const page = await newE2EPage();
    await page.setContent('<t-link>default</t-link>');
    const element = await page.find('t-link');
    expect(element).toHaveClass('hydrated');
  });
});
