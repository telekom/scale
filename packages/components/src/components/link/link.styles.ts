import { JssStyle } from 'jss';

export const styles: JssStyle = {
  link: {
    display: 'inline-flex',
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    position: 'relative',
    textDecoration: 'none',
    outline: 'none',
    cursor: 'pointer',
    padding: 0,
    fontSize: 14,
    fontWeight: 500,
    color: '#000',
    transition: 'all .2s ease-in-out',
    '--icon-color': '#000',
    '& scale-icon': {
      marginLeft: 4,
    },
    '&:hover&:not($link--disabled)': {
      color: 'black',
      '--icon-color': 'black',
    },
  },
  'link--underline': {
    '&:hover:not($link--disabled)': {
      textDecoration: 'underline',
    },
  },
  'link--disabled': {
    cursor: 'not-allowed',
    opacity: 0.5,
  },
  'link--variant-primary': {
    color: '#409eff',
    '--icon-color': '#409eff',
  },
  'link--variant-success': {
    color: '#67c23a',
    '--icon-color': '#67c23a',
  },
  'link--variant-warning': {
    color: '#e6a23c',
    '--icon-color': '#e6a23c',
  },
  'link--variant-danger': {
    color: '#f56c6c',
    '--icon-color': '#f56c6c',
  },
  'link--variant-info': {
    color: '#909399',
    '--icon-color': '#909399',
  },
};
