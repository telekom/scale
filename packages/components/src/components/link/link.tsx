import { Component, h, Prop } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-link',
  styleUrl: 'link.css',
  shadow: true,
})
export class Link {
  /** (optional) Link class */
  @Prop() public customClass?: string = '';
  /** (optional) Link href */
  @Prop() public href?: string = '';
  /** (optional) Disabled link */
  @Prop() public disabled?: boolean = false;
  /** (optional) Link underline */
  @Prop() public underline?: boolean = false;
  /** (optional) Link open a new tag */
  @Prop() public openNewTab?: boolean = false;
  /** (optional) Link variant */
  @Prop() public variant?: string = '';

  public render() {
    if (!!this.href && !this.disabled) {
      return (
        <a
          href={this.href}
          class={this.getCssClassMap()}
          target={this.openNewTab ? '_blank' : null}
        >
          <slot />
        </a>
      );
    }
    return (
      <div class={this.getCssClassMap()}>
        <slot />
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'link',
      this.customClass && this.customClass,
      this.disabled && `link--disabled`,
      this.underline && 'link--underline',
      this.variant && `link--variant-${this.variant}`
    );
  }
}
