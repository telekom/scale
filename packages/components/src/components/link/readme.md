# t-link



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute      | Description                    | Type      | Default |
| ------------ | -------------- | ------------------------------ | --------- | ------- |
| `disabled`   | `disabled`     | (optional) link disabled       | `boolean` | `false` |
| `href`       | `href`         | (optional) link href           | `string`  | `''`    |
| `openNewTab` | `open-new-tab` | (optional) link open a new tag | `boolean` | `false` |
| `underline`  | `underline`    | (optional) link variant        | `boolean` | `false` |
| `variant`    | `variant`      | (optional) link variant        | `string`  | `''`    |


## CSS Custom Properties

| Name                           | Description                       |
| ------------------------------ | --------------------------------- |
| `--link-align-items`           | Align items of the link           |
| `--link-color`                 | Color of the link                 |
| `--link-cursor`                | Cursor of the link                |
| `--link-display`               | Display of the link               |
| `--link-flex-direction`        | Flex direction of the link        |
| `--link-font-size`             | Font size of the link             |
| `--link-font-weight`           | Font weight of the link           |
| `--link-hover-with-underline`  | Color of the link variant primary |
| `--link-justify-content`       | Justify content of the link       |
| `--link-outline`               | Outline of the link               |
| `--link-padding`               | Padding of the link               |
| `--link-position`              | Position of the link              |
| `--link-text-decoration`       | Text decoration of the link       |
| `--link-variant-danger-color`  | Color of the link variant danger  |
| `--link-variant-info-color`    | Color of the link variant info    |
| `--link-variant-primary-color` | Color of the link variant primary |
| `--link-variant-success-color` | Color of the link variant success |
| `--link-variant-warning-color` | Color of the link variant warning |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
