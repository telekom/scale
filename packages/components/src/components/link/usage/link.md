## Default

<scale-link href="http://example.com">
	Label
</scale-link>

## Target

<scale-link href="http://example.com" target="_blank">
	Label
</scale-link>

## Disabled

<scale-link href="http://example.com" disabled>
	Label
</scale-link>

## Block

<scale-link href="http://example.com" block="true">
	Label
</scale-link>

## No underline

<scale-link href="http://example.com" underline="false">
	Label
</scale-link>

## Icon

<scale-link href="http://example.com" icon="M16.471,8.321 L12.613594,0.502886112 L8.755,8.321 L0.126435258,9.57533801 L6.37,15.66 L4.89610548,24.2548735 L12.613,20.197 L20.3310826,24.2548735 L18.856,15.66 L25.1007528,9.57533801 L16.471,8.321 Z M9.41891527,9.23556376 L12.613,2.761 L15.8082728,9.23556376 L22.951,10.273 L17.7826928,15.3122039 L19.003,22.427 L12.613594,19.067774 L6.223,22.427 L7.44449521,15.3122039 L2.275,10.273 L9.41891527,9.23556376 Z">
	Label
</scale-link>

## Icon size

<scale-link href="http://example.com" icon-size="16" icon="M16.471,8.321 L12.613594,0.502886112 L8.755,8.321 L0.126435258,9.57533801 L6.37,15.66 L4.89610548,24.2548735 L12.613,20.197 L20.3310826,24.2548735 L18.856,15.66 L25.1007528,9.57533801 L16.471,8.321 Z M9.41891527,9.23556376 L12.613,2.761 L15.8082728,9.23556376 L22.951,10.273 L17.7826928,15.3122039 L19.003,22.427 L12.613594,19.067774 L6.223,22.427 L7.44449521,15.3122039 L2.275,10.273 L9.41891527,9.23556376 Z">
	Label
</scale-link>
