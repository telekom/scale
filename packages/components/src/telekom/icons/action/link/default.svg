<svg width="24" height="24" xmlns="http://www.w3.org/2000/svg"><path d="M11.94 8.8c.665.255 1.29.645 1.82 1.18a5.245 5.245 0 01.18 7.235l-.18.19-3.535 3.535a5.215 5.215 0 01-3.71 1.535c-1.34 0-2.685-.51-3.71-1.535a5.26 5.26 0 01-.17-7.247l.17-.178L6.34 9.98c.27-.275.565-.505.875-.705a6.149 6.149 0 00.049 1.52l.056.325-3.455 3.455a3.76 3.76 0 000 5.305 3.76 3.76 0 005.15.146l.155-.146 3.535-3.535a3.755 3.755 0 000-5.305 3.857 3.857 0 00-.935-.68c-.05-.18-.08-.365-.08-.555 0-.355.095-.695.25-1.005zm1.825-6.24a5.26 5.26 0 017.425 0 5.242 5.242 0 01.162 7.246l-.172.179-3.535 3.535c-.27.275-.565.505-.875.705.04-.496.022-.989-.048-1.473l-.062-.362 3.465-3.465a3.76 3.76 0 000-5.305 3.736 3.736 0 00-2.65-1.095c-.886 0-1.772.311-2.478.933l-.172.162-3.535 3.535c-.71.705-1.1 1.65-1.1 2.65s.39 1.94 1.1 2.65c.28.28.595.505.93.68.065.25.1.505.075.77-.02.22-.085.5-.23.795a5.188 5.188 0 01-1.835-1.185 5.2 5.2 0 01-1.54-3.71c0-1.313.483-2.555 1.36-3.52l.18-.19 3.535-3.535z" fill="#000" fill-rule="evenodd"/></svg>