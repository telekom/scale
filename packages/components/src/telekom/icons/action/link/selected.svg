<svg width="24" height="24" xmlns="http://www.w3.org/2000/svg"><path d="M12.225 8.37a5.75 5.75 0 012.068 9.21l-.173.18-3.711 3.711a5.75 5.75 0 01-8.13-8.13L5.99 9.65a5.71 5.71 0 011.31-1 6.845 6.845 0 00-.105 1.155c0 .532.064 1.06.188 1.577l.082.308-3.245 3.24a3.255 3.255 0 004.449 4.732l.151-.132 3.535-3.535a3.26 3.26 0 00.135-4.456l-.259-.277a2.23 2.23 0 01-.118-2.748l.112-.144zm1.19-6.165a5.75 5.75 0 018.13 0 5.75 5.75 0 01.174 7.89l-.174.185-3.535 3.535a5.66 5.66 0 01-1.305 1c.02-.1.044-.211.059-.326l.011-.174a6.72 6.72 0 00-.163-2.194l-.082-.306 3.25-3.245a3.255 3.255 0 10-4.453-4.749l-3.682 3.684a3.26 3.26 0 00-.135 4.456l.259.278a2.25 2.25 0 01.016 2.896A5.64 5.64 0 019.88 13.87a5.745 5.745 0 01-.197-7.924l.197-.206 3.535-3.535z" fill="#000" fill-rule="evenodd"/></svg>