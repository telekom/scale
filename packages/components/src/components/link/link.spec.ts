import { newSpecPage } from '@stencil/core/testing';
import { Link } from './link';

describe('Link', () => {
  let element;
  beforeEach(async () => {
    element = new Link();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Link],
      html: `<t-link>default</t-link>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have class variant primary', () => {
    element.variant = 'primary';
    expect(element.getCssClassMap()).toContain('link--variant-primary');
  });

  it('should have class disabled ', () => {
    element.disabled = true;
    expect(element.getCssClassMap()).toContain('link--disabled');
  });

  it('should have class underline', () => {
    element.underline = true;
    expect(element.getCssClassMap()).toContain('link--underline');
  });

  it('should contain target="_blank" when openNewTab is set true', async () => {
    const page = await newSpecPage({
      components: [Link],
      html: `<t-link href="http://example.com" open-new-tab=true>Label</t-link>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should have link href value when href value is set', async () => {
    const page = await newSpecPage({
      components: [Link],
      html: `<t-link href="http://example.com">Label</t-link>`,
    });
    expect(page.root).toMatchSnapshot();
  });
});
