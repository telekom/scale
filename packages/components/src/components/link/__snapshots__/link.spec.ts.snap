// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should contain target="_blank" when openNewTab is set true 1`] = `
<scale-link href="http://example.com" open-new-tab="true">
  <mock:shadow-root>
    <style>
      .link-0-0-28 {
  color: #000;
  cursor: pointer;
  display: inline-flex;
  outline: none;
  padding: 0;
  position: relative;
  font-size: 16px;
  transition: all .2s ease-in-out;
  align-items: center;
  font-weight: 500;
  --icon-color: #000;
  flex-direction: row;
  justify-content: flex-start;
  text-decoration: none;
}
.link-0-0-28 scale-icon {
  margin-left: 4px;
}
.link-0-0-28:hover.link-0-0-28:not(.link--disabled-0-0-30) {
  color: black;
  --icon-color: black;
}
.link--underline-0-0-29 {}
.link--underline-0-0-29:hover:not(.link--disabled-0-0-30) {
  text-decoration: underline;
}
.link--disabled-0-0-30 {
  cursor: not-allowed;
  opacity: 0.5;
}
.link--block-0-0-31 {
  display: flex;
}
.link--variant-primary-0-0-32 {
  color: #409eff;
  --icon-color: #409eff;
}
.link--variant-success-0-0-33 {
  color: #67c23a;
  --icon-color: #67c23a;
}
.link--variant-warning-0-0-34 {
  color: #e6a23c;
  --icon-color: #e6a23c;
}
.link--variant-danger-0-0-35 {
  color: #f56c6c;
  --icon-color: #f56c6c;
}
.link--variant-info-0-0-36 {
  color: #909399;
  --icon-color: #909399;
}
    </style>
    <a class="link--underline-0-0-29 link-0-0-28" href="http://example.com" target="_self">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</scale-link>
`;

exports[`Link should have link href value when href value is set 1`] = `
<scale-link href="http://example.com">
  <mock:shadow-root>
    <style>
      .link-0-0-28 {
  color: #000;
  cursor: pointer;
  display: inline-flex;
  outline: none;
  padding: 0;
  position: relative;
  font-size: 16px;
  transition: all .2s ease-in-out;
  align-items: center;
  font-weight: 500;
  --icon-color: #000;
  flex-direction: row;
  justify-content: flex-start;
  text-decoration: none;
}
.link-0-0-28 scale-icon {
  margin-left: 4px;
}
.link-0-0-28:hover.link-0-0-28:not(.link--disabled-0-0-30) {
  color: black;
  --icon-color: black;
}
.link--underline-0-0-29 {}
.link--underline-0-0-29:hover:not(.link--disabled-0-0-30) {
  text-decoration: underline;
}
.link--disabled-0-0-30 {
  cursor: not-allowed;
  opacity: 0.5;
}
.link--block-0-0-31 {
  display: flex;
}
.link--variant-primary-0-0-32 {
  color: #409eff;
  --icon-color: #409eff;
}
.link--variant-success-0-0-33 {
  color: #67c23a;
  --icon-color: #67c23a;
}
.link--variant-warning-0-0-34 {
  color: #e6a23c;
  --icon-color: #e6a23c;
}
.link--variant-danger-0-0-35 {
  color: #f56c6c;
  --icon-color: #f56c6c;
}
.link--variant-info-0-0-36 {
  color: #909399;
  --icon-color: #909399;
}
    </style>
    <a class="link--underline-0-0-29 link-0-0-28" href="http://example.com" target="_self">
      <slot></slot>
    </a>
  </mock:shadow-root>
  Label
</scale-link>
`;

exports[`Link should match snapshot 1`] = `
<scale-link>
  <mock:shadow-root>
    <style>
      .link-0-0-10 {
  color: #000;
  cursor: pointer;
  display: inline-flex;
  outline: none;
  padding: 0;
  position: relative;
  font-size: 16px;
  transition: all .2s ease-in-out;
  align-items: center;
  font-weight: 500;
  --icon-color: #000;
  flex-direction: row;
  justify-content: flex-start;
  text-decoration: none;
}
.link-0-0-10 scale-icon {
  margin-left: 4px;
}
.link-0-0-10:hover.link-0-0-10:not(.link--disabled-0-0-12) {
  color: black;
  --icon-color: black;
}
.link--underline-0-0-11 {}
.link--underline-0-0-11:hover:not(.link--disabled-0-0-12) {
  text-decoration: underline;
}
.link--disabled-0-0-12 {
  cursor: not-allowed;
  opacity: 0.5;
}
.link--block-0-0-13 {
  display: flex;
}
.link--variant-primary-0-0-14 {
  color: #409eff;
  --icon-color: #409eff;
}
.link--variant-success-0-0-15 {
  color: #67c23a;
  --icon-color: #67c23a;
}
.link--variant-warning-0-0-16 {
  color: #e6a23c;
  --icon-color: #e6a23c;
}
.link--variant-danger-0-0-17 {
  color: #f56c6c;
  --icon-color: #f56c6c;
}
.link--variant-info-0-0-18 {
  color: #909399;
  --icon-color: #909399;
}
    </style>
    <a class="link--underline-0-0-11 link-0-0-10" target="_self">
      <slot></slot>
    </a>
  </mock:shadow-root>
  default
</scale-link>
`;
